control/state.py

```diff
@@ -185,21 +185,33 @@ def reset(self, omap_state):
         """Resets dictionary with OMAP state."""
         self.state = omap_state
 
+class ReleasedLock:
```

Comment:
This class seems like an overkill. I think it's much clearer to have the release and acquire calls in the code then using this class. Especially when it acts opposite to intuition, releasing when you go in and locking when you go out.

Reply:
@gbregman employing a context manager is crucial in this context. the objective is to guarantee lock acquisition even if an exception is raised within the block where the lock is released (in our case, during a sleep operation). Resorting to a release/acquire mechanism wouldn't be suitable in such scenarios.